src/app/data-entries/dropdown/dropdown-tree-hierarchical-selection/dropdown-tree-hierarchical-selection.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { IBaseChipEventArgs, IgxTreeComponent, ITreeNodeSelectionEvent } from 'igniteui-angular';
+import { SINGERS } from '../../../data/singersData';
```

Comment:
For this sample, use different kind of data. The data is not homogeneous and that's why there was custom logic to not take text from every selected node. Instead, the data should be homogeneous and the selection shouldn't cascade, so it better displays a real-world scenario :)